### PR TITLE
feature(oidc):add oidc

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/PersistedGrantModel.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    /// <summary>
+    /// Gets or sets the data.
+    /// </summary>
+    /// <value>
+    /// The data.
+    /// </value>
+    public string Data { get; set; }
+
+    public PersistedGrantModel(string key, string type, string subjectId, string sessionId, string clientId, string description, DateTime creationTime, DateTime? expiration, DateTime? consumedTime, string data)
```
Comment (zhenlei520) [inline src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/PersistedGrantModel.cs:91]: line break recommended

File: src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/IdentityResourceModel.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Name = name;
+        DisplayName = displayName;
+        UserClaims = userClaims;
+    }
+
+    public IdentityResourceModel(string name, ICollection<string> userClaims) : this(name, name, userClaims)
+    {
+    }
+
+    public IdentityResourceModel(string name, string displayName, string? description, bool enabled, bool showInDiscoveryDocument, bool required, bool emphasize, ICollection<string> userClaims, IDictionary<string, string>? properties) : this(name, displayName, userClaims)
```
Comment (zhenlei520) [inline src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/IdentityResourceModel.cs:29]: line break recommended

File: src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/ClientClaimModel.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    /// <summary>
+    /// The claim value type
+    /// </summary>
+    public string ValueType { get; set; } = ClaimValueTypes.String;
+
+    /// <summary>
+    /// ctor
+    /// </summary>
+    /// <param name="type"></param>
+    /// <param name="value"></param>
```
Comment (zhenlei520) [inline src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/ClientClaimModel.cs:27]: If there is no description, it is recommended to remove

File: src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/IdentityResourceModel.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        Description = description;
+        Enabled = enabled;
+        ShowInDiscoveryDocument = showInDiscoveryDocument;
+        Required = required;
+        Emphasize = emphasize;
+        Properties = properties;
```
Comment (zhenlei520) [inline src/Oidc/Masa.BuildingBlocks.Oidc.Models/Models/IdentityResourceModel.cs:36]: Properties is a non-null value

File: src/Oidc/Masa.BuildingBlocks.Oidc.Domain/Repositories/IRepositoryBase.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Oidc.Domain.Repositories;
+
+public interface IRepositoryBase<TEntity> where TEntity : class, IEntity
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Oidc/Masa.BuildingBlocks.Oidc.Domain/Repositories/IRepositoryBase.cs:6]: @doddgu At present, what we define starts with Base, but it is better for some services to end with Base ( `ServiceBase`, `ControllerBase` ), does this need to be unified again?

**Review by zhenlei520 — APPROVED** [review]
@doddgu 
LGTM

File: src/Authentication/Masa.BuildingBlocks.Authentication.Oidc.Models/Models/GrantType.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Authentication.Oidc.Models.Models;
+
+public class GrantType
+{
+    public const string Implicit = "implicit";
```
Comment (doddgu) [inline src/Authentication/Masa.BuildingBlocks.Authentication.Oidc.Models/Models/GrantType.cs:8]: Upper.

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**